### ISSUE TITLE
SBO terms not exported properly to Excel format when they are within MIRIAM annotations

### ISSUE THREAD

**Issue by pranasag** [SysBioChalmers/RAVEN#516]
### Description of the issue:
Hi everyone, 

I have this issue on exporting models to Excel format. Systems Biology Ontology (SBO) terms are not properly handled in the `exportToExcelFormat()` function, when they are part of MIRIAM annotations. These terms get exported as `sbo/SBO:-1.000000e+00`. The SBO terms are represented correctly in the loaded model object in RAVEN environment.

#### Reproducing this issue:
See a snippet of the raw SBML model to get the feeling:

```xml
      <fbc:geneProduct metaid="meta_<geneId>" sboTerm="SBO:0000243" fbc:id="<geneId>" fbc:label="<geneName>">
        <annotation>
          <rdf:RDF xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#" xmlns:dcterms="http://purl.org/dc/terms/" xmlns:vCard="http://www.w3.org/2001/vcard-rdf/3.0#" xmlns:vCard4="http://www.w3.org/2006/vcard/ns#" xmlns:bqbiol="http://biomodels.net/biology-qualifiers/" xmlns:bqmodel="http://biomodels.net/model-qualifiers/">
            <rdf:Description rdf:about="#meta_<geneId>">
              <bqbiol:is>
                <rdf:Bag>
                  <rdf:li rdf:resource="http://identifiers.org/ncbiprotein/<geneId>"/>
                  <rdf:li rdf:resource="http://identifiers.org/sbo/SBO:0000243"/>
                </rdf:Bag>
              </bqbiol:is>
            </rdf:Description>
          </rdf:RDF>
        </annotation>
      </fbc:geneProduct>
```
If the SBO term is defined in the highest-level clause (`fbc:geneProduct metaid="meta_<geneId>" sboTerm="SBO:0000243"`), it's processed properly, but if it falls under annotations (`<rdf:li rdf:resource="http://identifiers.org/sbo/SBO:0000243"/>`) it throws `sbo/SBO:-1.000000e+00` in the Excel sheet. Here, gene is taken as an example, but the same happens to reactions and metabolites.

#### System information
*** THE RAVEN TOOLBOX ***

 > Installing from location             /Users/pranas/Documents/MATLAB/RAVEN
 > Checking RAVEN release               2.8.6
   > You are running the latest RAVEN release
 > Checking MATLAB release              2023b
 > Checking system architecture         maci64
 > Set RAVEN in MATLAB path             Pass
 > Save MATLAB path                     Pass
   > Make binaries executable           Pass

=== Model import and export ===
 > Add Java paths for Excel format      Pass
 > Checking libSBML version             5.19.0
 > Checking model import and export
   > Import Excel format                Pass
   > Export Excel format                Pass
   > Import SBML format                 Pass
   > Export SBML format                 Pass

=== Model solvers ===
 > Checking for LP solvers
   > glpk                               Pass
   > gurobi                             Pass
   > soplex                             Fail
   > cobra                              Fail
 > Set RAVEN solver                     glpk

=== Essential binary executables ===
 > Checking BLAST+                      Pass
 > Checking DIAMOND                     Pass
 > Checking HMMER                       Pass

=== Compatibility ===
 > Checking function uniqueness         Pass

**I hereby confirm that I have:**
- [X] Checked that a similar issue does not [already exist](https://github.com/SysBioChalmers/RAVEN/issues?utf8=%E2%9C%93&q=is%3Aissue)

**Comment by edkerk** [comment]
Thank you for your patience. The problem is actually in importModel: if no `sboTerm` is provided in fbc object in SBML, `TranslateSBML` gives it the value -1, which becomes problematic when SBO:0000x is prefixed at a later stage during import.
This is now fixed in e41c0786dff4fb862ec75246fe8b7475a9016095.

Then, another issue became apparent. Currently, `importModel` does not read SBO terms if specified as identifiers.org within `<annotation>`. Likewise, `exportModel` writes SBO terms into `sboTerms`, not in `<annotation>`. This is not only true for genes, but also for `species` and `reactions`. Do you have a particular reason why SBO term are specified as identifiers.org? Should this be considered instead of `sboTerm`? IMHO, `sboTerm` should then have precedence.

**Comment by edkerk** [comment]
Indeed, the current behaviour is to fully ignore SBO terms in the MIRIAM block.

I can easily change this to the following: (1) import SBO terms from the MIRIAM block; (2) overwrite these SBO terms if a valid `sboTerm` is present. So the SBML object header would still have priority.

**Comment by pranasag** [comment]
I am now looking at the [latest SBML specifications](https://raw.githubusercontent.com/combine-org/combine-specifications/main/specifications/files/sbml.level-3.version-2.core.release-2.pdf), and pt. 5.2.2 (p. 97) suggests these terms should be unique for a single object. Perhaps also relevant to implement a check whether # of SBO terms <= 1 (if not done already)?

**Comment by edkerk** [comment]
Good point, now included in PR #536
